knowledge_agent/github.py

```
@@ -0,0 +1,215 @@
+    @staticmethod
+    def fetch_latest_commit_sha(token: str, owner: str, repo: str, branch: Optional[str] = None) -> Optional[str]:
+        """Fetches the latest commit SHA for the target or default branch."""
+        target_branch = branch or "main"
+        try:
+            with httpx.Client(timeout=10.0) as client:
+                res = client.get(f"{GITHUB_API_BASE}/repos/{owner}/{repo}/commits/{target_branch}", headers=GitHubClient._get_headers(token))
+                if res.status_code != 200 and target_branch != "master":
+                    res = client.get(f"{GITHUB_API_BASE}/repos/{owner}/{repo}/commits/master", headers=GitHubClient._get_headers(token))
+                if res.status_code == 200:
+                    return res.json().get("sha", "")[:40]
+        except Exception as e:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)